lib/govuk_app_config/govuk_error.rb

```diff
@@ -12,8 +13,6 @@ def self.notify(exception_or_message, args = {})
   end
 
   def self.configure
-    Raven.configure do |config|
-      yield(config)
-    end
+    yield Configuration.new(Raven.configuration)
```

Comment:
I think configuration would benefit from being saved as a variable - otherwise if you run configure twice it'll keep all the Raven configuration but lose any of our custom configuration.

Reply:
I could change it to this:

```ruby
  def self.configure
    @configuration ||= Configuration.new(Raven.configuration)
    yield @configuration
  end
```

However, it is easier said than done - I have the same problematic test [as below](https://github.com/alphagov/govuk_app_config/pull/160#discussion_r504475270). By the time this test is run, the `environment` has already been set and is no longer looking at the `SENTRY_CURRENT_ENV` ENV variable. I'll need to have a think about how to get this test working.

Reply:
Oh sorry should have read this comment before commenting on the last one - oh well. It now makes sense to me.

So behaviour wise setting the instance variable seems the right thing to do but yes all class instance variables are a real pain with testing.

Given we have the same singleton behaviour with raven I think it might be better to rethink the testing strategy to accommodate it. So we could test that it yields a configuration instance? After a [quick google](https://relishapp.com/rspec/rspec-expectations/docs/built-in-matchers/yield-matchers) perhaps it could be `expect { |b| GovukError.configure(&b) }.to yield_once_with_args(instance_of(GovukError::Configuration))`

Reply:
Thanks Kevin - I hadn't found time to look at this earlier, but have now implemented the test change. I've encapsulated it in its own commit at the end of the PR.